code/mathematical-algorithms/factorial/factorialstirling.c

```diff
@@ -0,0 +1,27 @@
+// Factorial using Stirling's approximation in c language
+
+float stirling_approximation(int n) {
+    int fact;
+    float stirling, ans;
+    fact = factorial(n);
+    stirling = sqrt(2.0*3.1416*n) * pow(n / 2.71828, n);
+    ans = fact / stirling;
```

Comment:
The Stirling's approximation seems wrong.
It should be `pow(exp,-n) * pow(n,n) * sqrt(2*n*PI) ;`
In you case, it's n! / (sqrt(2*n*PI) * pow(exp, -n) * n^n)
You can use the input as 10 to check its value, it should close to 3628800.